compiler/z/codegen/OMRInstruction.cpp

```diff
@@ -683,37 +700,50 @@ static void handleLoadWithRegRanges(TR::Instruction *inst, TR::CodeGenerator *cg
          opCodeToUse == TR::InstOpCode::LMY || opCodeToUse == TR::InstOpCode::VLM))
       return;
 
-   bool isVector = opCodeToUse == TR::InstOpCode::VLM ? true : false;
-   uint32_t numVRFs = TR::RealRegister::LastAssignableVRF - TR::RealRegister::FirstAssignableVRF;
+   TR::Register *lowReg, *highReg;
+   uint32_t minRegNum, maxRegNum;
+   uint32_t availForShuffle;
 
-   TR::Register *lowReg  = isVector ? toS390VRSInstruction(inst)->getFirstRegister()  : toS390RSInstruction(inst)->getFirstRegister();
-   TR::Register *highReg = isVector ? toS390VRSInstruction(inst)->getSecondRegister() : toS390RSInstruction(inst)->getSecondRegister();
+   if (opCodeToUse == TR::InstOpCode::VLM)
+      {
+      lowReg = toS390VRSInstruction(inst)->getFirstRegister();
+      highReg = toS390VRSInstruction(inst)->getSecondRegister();
+      minRegNum = TR::RealRegister::FirstAssignableVRF;
+      maxRegNum = TR::RealRegister::LastAssignableVRF;
+      availForShuffle = cg->machine()->genBitMapOfAssignableVRFs() & ~(getRegMaskFromRange(inst, cg, true));
+      }
+   else
+      {
+      lowReg = toS390RSInstruction(inst)->getFirstRegister();
+      highReg = toS390RSInstruction(inst)->getSecondRegister();
+      minRegNum = TR::RealRegister::FirstGPR;
+      maxRegNum = TR::RealRegister::LastAssignableGPR;
+      availForShuffle = cg->machine()->genBitMapOfAssignableGPRs() & ~(getRegMaskFromRange(inst, cg, false));
+      }
 
    if (inst->getRegisterOperand(1)->getRegisterPair())
       highReg = inst->getRegisterOperand(1)->getRegisterPair()->getHighOrder();
-   uint32_t lowRegNum = ANYREGINDEX(toRealRegister(lowReg)->getRegisterNumber());
-   uint32_t highRegNum = ANYREGINDEX(toRealRegister(highReg)->getRegisterNumber());
+   uint32_t lowRegNum = toRealRegister(lowReg)->getRegisterNumber();
+   uint32_t highRegNum = toRealRegister(highReg)->getRegisterNumber();
 
    uint32_t regsInRange;
    if (highRegNum >= lowRegNum)
       regsInRange = (highRegNum - lowRegNum + 1);
    else
-      regsInRange = ((isVector ? numVRFs : 16) - lowRegNum) + highRegNum + 1;
+      regsInRange = ((maxRegNum + 1) - lowRegNum) + highRegNum + 1;
```

Comment:
Seems like it will calculate incorrect range (offset by plus 1) in case of GPRs (Again as I mentioned in previous comment, we should not have lowRegNum higher than highRegNum if the instruction is Vector Load Multiple.)